src/Rule/Rule.php

```diff
@@ -117,13 +117,21 @@ public static function parse(ParserState $oParserState): Rule
      * @param string $sRule
      *
      * @return array<int, string>
```

Comment:
Should we make this more specific as we're touching the return value anyway?

```suggestion
     * @return list<non-empty-string>
```

Reply:
Done.  Also noticed that a PHP type hint for the parameter could be used, but I think that should be a separate PR (particularly as we may want to backport retaining support for a PHP version which doesn't allow argument type hints for primitive types).